smithy-syntax/src/main/java/software/amazon/smithy/syntax/SortUseStatements.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.syntax;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import software.amazon.smithy.model.loader.IdlToken;
+
+/**
+ * Sorts use statements in models alphabetically, case-sensitively.
+ */
+final class SortUseStatements implements Function<TokenTree, TokenTree> {
+    // One of the complications of this transformation is that comments that might document a use statement are
+    // captured in the BR production that trails a use statement. So moving a use statement needs to move any trailing
+    // comments from its previous sibling to the next previous sibling it's moved after. To make this even more
+    // complicated, comments on the same line as a statement should stay with the statement rather than moved.
+    @Override
+    public TokenTree apply(TokenTree tree) {
+        TreeCursor useSection = Objects.requireNonNull(tree.zipper().getFirstChild(TreeType.SHAPE_SECTION))
+                .getFirstChild(TreeType.USE_SECTION);
+
+        // All trees should have a USE_SECTION. If the section is empty, then no use statements need to be sorted.
+        // Note that if it isn't empty, then there is a guaranteed NAMESPACE_STATEMENT too.
+        if (useSection == null || useSection.getTree().isEmpty()) {
+            return tree;
+        }
```

Comment:
Should we have a test case for this situation?

Reply:
Most test cases don't have any use statements in them, so it's covered